httpu/httpu.go

```diff
@@ -26,6 +27,22 @@ type ClientInterface interface {
 	) ([]*http.Response, error)
 }
 
+// ClientInterfaceCtx is the equivalent of ClientInterface, except with methods
+// taking a context.Context parameter.
+type ClientInterfaceCtx interface {
+	// DoWithContext performs a request. If the input request has a
+	// deadline, then that value will be used as the timeout for how long
+	// to wait before returning the responses that were received. If the
```

Comment:
IIUC, there's an important caveat to mention here: if the context isn't cancelled, nor has a deadline, then this function will ~never return (making the call useless, as it won't return any values).

If I'm correct in that, then it should be made very clear in this documentation that the caller _must_ call with a `req` that has a context that itself is to be cancelled or has a deadline/timeout.

Reply:
Added a comment to that effect; does that look good?

Reply:
LGTM, thanks!